File: c/estruturas_repetidas/dentro_fora.c
/* Problema "dentro_fora"
 Leia um valor inteiro N. Este valor será a quantidade de valores inteiros X que serão lidos em seguida.
 Mostre quantos destes valores X estão dentro do intervalo [10,20] e quantos estão fora do intervalo,
 conforme exemplo
*/

#include <stdio.h>

int main()
{
  int n, x, inside = 0, outside = 0;
  const int LOWER_BOUND = 10;
  const int UPPER_BOUND = 20;

  printf("Quantos números você vai digitar? ");
  scanf("%d", &n);

  for (int i = 0; i < n; i++)
  {
    printf("Digite um número: ");
    scanf("%d", &x);

    if (LOWER_BOUND <= x && x <= UPPER_BOUND)
    {
      inside++;
    }
    else
    {
      outside++;
    }
  }

  printf("%d DENTRO\n", inside);
  printf("%d FORA\n", outside);

  return 0;
}

File: java/src/estruturas/repetidas/DentroFora.java
/*
 */

package estruturas.repetidas;

import java.util.Scanner;

/**
 * Problema "dentro_fora"
 * Leia um valor inteiro N.
 * Este valor será a quantidade de valores inteiros X que serão lidos em seguida.
 * Mostre quantos destes valores X estão dentro do intervalo [10,20]
 * e quantos estão fora do intervalo.
 */
public class DentroFora {
  private static final int LOWER_BOUND = 10;
  private static final int UPPER_BOUND = 20;
  private static final int MAX_NUMBERS = 10;

  /**
   * The entry point of application.
   *
   * @param args the input arguments
   */
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    System.out.printf("Quantos números você vai digitar (máximo %d)? ", MAX_NUMBERS);
    int n = sc.nextInt();

    n = n < 1 || n > MAX_NUMBERS ? 1 : n;

    int x;
    int inside = 0;
    int outside = 0;

    for (int i = 0; i < n; i++) {
      System.out.print("Digite um número: ");
      x = sc.nextInt();

      if (LOWER_BOUND <= x && x <= UPPER_BOUND) {
        inside++;
      } else {
        outside++;
      }
    }

    System.out.printf("%d DENTRO%n", inside);
    System.out.printf("%d FORA%n", outside);

    sc.close();
  }
}